backend_tv/adjust_src.cpp

```diff
@@ -481,4 +481,43 @@ void fixupOptimizedTgt(llvm::Function *tgt) {
     myAllocCall->eraseFromParent();
 }
 
+/**
+ * Given an intToPtr instruction, checks for a round trip 
+ * from a ptrToInt instruction, then replaces with a single GEP instruction.
+ * This translation can be verified with physical pointers.
+ */
+void tryReplaceRoundTrip(llvm::IntToPtrInst *intToPtr) {
+  if (!intToPtr)
+    return;
+
+  // we only understand add instructions in this context
+  auto *op_inst = dyn_cast<llvm::Instruction>(intToPtr->getOperand(0));
+  if (!op_inst || op_inst->getOpcode() != llvm::Instruction::Add || op_inst->getNumUses() > 1)
+      return;
+
+  // keep track of (ptr + int) vs (int + ptr)
+  bool ptrOnLeft = true;
```

Comment:
ok, since you have two distinct cases in your optimization, it would be best to add two test cases, one corresponding to each thing that can happen here